crates/tiny/src/main.rs

```diff
@@ -183,3 +198,30 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth<String>) -> Result<SASLAuth, String> {
```

Comment:
We could do error checking in this function in `Config::validate`, but that would mean exiting in case of an error (instead of connecting without SASL authentication). Not sure which behavior is better..

I think probably the tls checking should be done in `Config::validate` as if that check fails that means a configuration error.

Reply:
Good idea. I think exiting is better. Moving it there.

Reply:
@osa1 Just to be clear, we still want to check the file read error at runtime, right? e.g display to the user in the server tab that the PEM file cannot be read.